bin/agent-data-plane/src/run_subcommand.rs

```diff
@@ -0,0 +1,241 @@
+use std::time::{Duration, Instant};
```

Comment:
I would move this -- and the other subcommand-specific files -- to a new directory, like `cli` or something.

So this would become `bin/agent-data-plane/src/cli/run.rs`, the debug stuff would be `bin/agent-data-plane/src/cli/debug.rs`, and so on.